frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,243 @@
-import * as React from 'react';
-import {
-  Button,
-  Form,
-  FormGroup,
-  FormHelperText,
-  HelperText,
-  HelperTextItem,
-  Label,
-  LabelGroup,
-  Modal,
-  TextInput,
-} from '@patternfly/react-core';
-import { ExclamationCircleIcon } from '@patternfly/react-icons';
-import DashboardDescriptionListGroup, {
-  DashboardDescriptionListGroupProps,
-} from './DashboardDescriptionListGroup';
-
-type EditableTextDescriptionListGroupProps = Partial<
-  Pick<DashboardDescriptionListGroupProps, 'title' | 'contentWhenEmpty'>
-> & {
+import React, { useState } from 'react';
+import { Label, LabelGroup, Alert, AlertVariant, Button } from '@patternfly/react-core';
+import DashboardDescriptionListGroup from './DashboardDescriptionListGroup';
+
+interface EditableLabelsProps {
   labels: string[];
-  saveEditedLabels: (labels: string[]) => Promise<unknown>;
-  allExistingKeys?: string[];
+  onLabelsChange: (labels: string[]) => Promise<void>;
   isArchive?: boolean;
-};
+  title?: string;
+  contentWhenEmpty?: string;
+  allExistingKeys: string[];
+}
 
-const EditableLabelsDescriptionListGroup: React.FC<EditableTextDescriptionListGroupProps> = ({
+export const EditableLabelsDescriptionListGroup: React.FC<EditableLabelsProps> = ({
   title = 'Labels',
   contentWhenEmpty = 'No labels',
   labels,
-  saveEditedLabels,
+  onLabelsChange,
   isArchive,
-  allExistingKeys = labels,
+  allExistingKeys,
 }) => {
-  const [isEditing, setIsEditing] = React.useState(false);
-  const [unsavedLabels, setUnsavedLabels] = React.useState(labels);
-  const [isSavingEdits, setIsSavingEdits] = React.useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isSavingEdits, setIsSavingEdits] = useState(false);
+  const [unsavedLabels, setUnsavedLabels] = useState(labels);
 
-  const editUnsavedLabel = (newText: string, index: number) => {
-    if (isSavingEdits) {
+  const validateLabels = (labelsList: string[]): string[] => {
+    const errors: string[] = [];
+
+    const duplicatesMap = new Map<string, number>();
+    labelsList.forEach((label) => {
+      duplicatesMap.set(label, (duplicatesMap.get(label) || 0) + 1);
+    });
+
+    const duplicateLabels: string[] = [];
+    duplicatesMap.forEach((count, label) => {
+      if (count > 1) {
+        duplicateLabels.push(label);
```

Comment:
@YuliaKrimerman I hate to drag this out a bit longer but can we add a test or two that covers the duplicate label validation to make codecov happier?